### DISCUSSION TITLE
How to display the frame of each joint in pybullet

### DISCUSSION THREAD

**Discussion by huxiaoyi0625** [bulletphysics/bullet3#3867] [Q&A] [answered]
I want to display the frame of each joint so that I can debug the simulation of my robot better. However, I found that the frame of joints seems different from the results shown in the RVIZ, and it gives me big trouble. What should I do?

**Comment by cvshah** [comment]
hello, I wrote a short function for this. I hope you find this useful.

def axiscreator(bodyId, linkId = -1):
    print(f'axis creator at bodyId = {bodyId} and linkId = {linkId} as XYZ->RGB')
    x_axis = self.minitaur._pybullet_client.addUserDebugLine(lineFromXYZ          = [0, 0, 0]  ,
                                                               lineToXYZ            = [0.1, 0, 0],
                                                               lineColorRGB         = [1, 0, 0]  ,
                                                               lineWidth            = 0.1        ,
                                                               lifeTime             = 0          ,
                                                               parentObjectUniqueId = bodyId     ,
                                                               parentLinkIndex      = linkId     )
      
     y_axis = self.minitaur._pybullet_client.addUserDebugLine(lineFromXYZ          = [0, 0, 0]  ,
                                                               lineToXYZ            = [0, 0.1, 0],
                                                               lineColorRGB         = [0, 1, 0]  ,
                                                               lineWidth            = 0.1        ,
                                                               lifeTime             = 0          ,
                                                               parentObjectUniqueId = bodyId     ,
                                                               parentLinkIndex      = linkId     )

     z_axis = self.minitaur._pybullet_client.addUserDebugLine(lineFromXYZ          = [0, 0, 0]  ,
                                                               lineToXYZ            = [0, 0, 0.1],
                                                               lineColorRGB         = [0, 0, 1]  ,
                                                               lineWidth            = 0.1        ,
                                                               lifeTime             = 0          ,
                                                               parentObjectUniqueId = bodyId     ,
                                                               parentLinkIndex      = linkId     )
    return [x_axis, y_axis, z_axis]

**Reply by huxiaoyi0625** [reply]
Thanks for your great help！ I want to show the URDF link frame.  Although the code couldn't thoroughly solve my problem,  you inspired me with the idea of how to solve it!

**Comment by cadop** [comment, marked as answer]
I think if you use the visualizer in pybullet, pressing `w` makes it wireframe, and `j` shows the axis.

**Reply by cvshah** [reply]
wow, it works. Thank you

**Reply by huxiaoyi0625** [reply]
Thanks for your great help!
